### PR TITLE
[BACKLOG-24042] FTPS GET data channel encryption support for implicit TLS/SSL

### PR DESCRIPTION
**Pull request by tgf**

### CODE REVIEW

**Comment by tgf** [comment]
@pentaho/kashyyyk please review

File: engine/src/main/java/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java [hunk 1]
```diff
@@ -169,6 +173,11 @@ public void connect() throws KettleException {
       connection =
         FTPConnectionFactory.getInstance( getProperties(
           hostName, portNumber, userName, passWord, connectionType, timeOut, passiveMode ) );
+      if ( connection.getConnectionType() == FTPConnection.IMPLICIT_SSL_WITH_CRYPTED_DATA_FTP_CONNECTION
```
Comment (joaormf) [inline engine/src/main/java/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java]: Is there any resources that the previous connection object could release?

File: engine/src/main/java/org/pentaho/di/job/entries/ftpsget/ftp4che/SecureDataFTPConnection.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    setConnectionStatusLock( CSL_INDIRECT_CALL );
+    setConnectionStatus( FTPConnection.BUSY );
+
+    ListCommand command = new ListCommand( directory );
+    SocketProvider provider = null;
+    String workDirectory = getWorkDirectory();
+
+    if ( isCryptedDataConnection() ) {
+      Command pbsz = new Command( Command.PBSZ, "0" );
+      ( sendCommand( pbsz ) ).dumpReply();
```
Comment (joaormf) [inline engine/src/main/java/org/pentaho/di/job/entries/ftpsget/ftp4che/SecureDataFTPConnection.java]: why the extra parenthesis? and couldn't we send the command, and then call super.getDirectoryListing(directory)?

File: engine/src/main/java/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java (hunk 1 above)
Comment (tgf) [reply engine/src/main/java/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java]: not before connect()

File: engine/src/main/java/org/pentaho/di/job/entries/ftpsget/ftp4che/SecureDataFTPConnection.java (hunk 2 above)
Comment (tgf) [reply engine/src/main/java/org/pentaho/di/job/entries/ftpsget/ftp4che/SecureDataFTPConnection.java]: sending the command before sounds like a better idea..leave this on hold

**Comment by tgf** [comment]
@pentaho/kashyyyk updated, please review

File: engine/src/test/java/org/pentaho/di/job/entries/ftpsget/FTPSConnectionTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.job.entries.ftpsget;
+
+import static org.junit.Assert.*;
```
[the remark this answers is not in the document]
Comment (tgf) [reply engine/src/test/java/org/pentaho/di/job/entries/ftpsget/FTPSConnectionTest.java]: it's a test

**Pull request merged**